assets/tchatcha/codes.php

```diff
@@ -486,4 +486,4 @@ private function binhex($bin)
 		return strrev($hex);
 	}
 
-?>
+}
```

Comment:
Ending with `?>` (plus a newline) will cause issues if this script is included anywhere where the HTTP response has not been sent yet, so this can prevent creating sessions or redirecting.